ruby/ql/lib/codeql/xml/XML.qll

```diff
@@ -0,0 +1,307 @@
+/**
+ * Provides classes and predicates for working with XML files and their content.
```

Comment:
This seems like another candidate for a shared library pack, but that can be done separately.

Reply:
I actually copied the XML.qll files that the other languages share via `identical-files.json` . Unfortunately, the shared file contains an invalid (for us) `import semmle.files.FileSystem` statement. In addition the shared file has very many deprecated, renamed classes. So I cleaned it up.

Ideally we make this a proper shared library pack based on @erik-krogh's work on a shared Locations/Files library.

Reply:
> Ideally we make this a proper shared library pack based on @erik-krogh's work on a shared Locations/Files library. 

I think it will take a while before I got the CLI features I need to make that library. So don't wait on that.